From 760c01673f1887a3cd70b8feff52c39991a1e03f Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 10 May 2021 11:21:50 +0300
Subject: [PATCH] ucm-imx8m-plus: Update boot environment

Replace all static variable with environment variables.
For future modifications if required.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index a53ca36490..e08c4d8b94 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -134,13 +134,20 @@
 				"echo WARN: Cannot load the DT; " \
 			"fi; " \
 		"fi;\0" \
+		"emmc_root=/dev/mmcblk2p2\0" \
+		"emmc_dev=2\0" \
+		"sd_root=/dev/mmcblk1p2\0" \
+		"sd_dev=1\0" \
+		"usb_root=/dev/sda2\0" \
+		"usb_dev=0\0" \
+		"boot_part=1\0" \
 		"root_opt=rootwait rw\0" \
-		"emmc_ul=setenv iface mmc; setenv dev 2; setenv part 1;" \
-		"setenv bootargs console=${console} root=/dev/mmcblk2p2 ${root_opt};\0" \
-		"sd_ul=setenv iface mmc; setenv dev 1; setenv part 1;" \
-			"setenv bootargs console=${console} root=/dev/mmcblk1p2 ${root_opt};\0" \
-		"usb_ul=usb start; setenv iface usb; setenv dev 0; setenv part 1;" \
-			"setenv bootargs console=${console} root=/dev/sda2 ${root_opt};\0" \
+		"emmc_ul=setenv iface mmc; setenv dev ${emmc_dev}; setenv part ${boot_part};" \
+		"setenv bootargs console=${console} root=${emmc_root} ${root_opt};\0" \
+		"sd_ul=setenv iface mmc; setenv dev ${sd_dev}; setenv part ${boot_part};" \
+			"setenv bootargs console=${console} root=${sd_root} ${root_opt};\0" \
+		"usb_ul=usb start; setenv iface usb; setenv dev ${usb_dev}; setenv part ${boot_part};" \
+			"setenv bootargs console=${console} root=${usb_root} ${root_opt};\0" \
 		"ulbootscript=load ${iface} ${dev}:${part} ${loadaddr} ${script};\0" \
 		"ulimage=load ${iface} ${dev}:${part} ${loadaddr} ${image}\0" \
 		"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-- 
2.11.0

